pkg/controller/network/vpc/controller.go

```diff
@@ -97,131 +116,127 @@ type external struct {
 		VpcIds: []string{meta.GetExternalName(cr)},
-	})
-	rsp, err := req.Send(ctx)
-	if ec2.IsVPCNotFoundErr(err) {
-		return managed.ExternalObservation{ResourceExists: false}, nil
-	}
+	}).Send(ctx)
+
 	if err != nil {
-		return managed.ExternalObservation{}, errors.Wrap(err, errDescribe)
+		return managed.ExternalObservation{}, errors.Wrapf(resource.Ignore(ec2.IsVPCNotFoundErr, err), errDescribe)
 	}
 
 	// in a successful response, there should be one and only one object
-	if len(rsp.Vpcs) != 1 {
+	if len(response.Vpcs) != 1 {
 		return managed.ExternalObservation{}, errors.New(errMultipleItems)
 	}
 
-	observed := rsp.Vpcs[0]
+	observed := response.Vpcs[0]
 
 	if observed.State == awsec2.VpcStateAvailable {
 		cr.SetConditions(runtimev1alpha1.Available())
+	} else if observed.State == awsec2.VpcStatePending {
+		cr.SetConditions(runtimev1alpha1.Creating())
 	}
 
-	cr.UpdateExternalStatus(observed)
+	// update the CRD spec for any new values from provider
+	current := cr.Spec.ForProvider.DeepCopy()
+	ec2.LateInitializeVPC(&cr.Spec.ForProvider, &observed)
+	if !cmp.Equal(current, &cr.Spec.ForProvider) {
+		if err := e.kube.Update(ctx, cr); err != nil {
+			return managed.ExternalObservation{}, errors.Wrap(err, errSpecUpdate)
+		}
+	}
+	cr.Status.AtProvider = ec2.GenerateVpcObservation(observed)
 
 	return managed.ExternalObservation{
-		ResourceExists:    true,
-		ConnectionDetails: managed.ConnectionDetails{},
-		ResourceUpToDate:  ec2.IsUpToDate(cr.Spec.VPCParameters, observed),
+		ResourceExists:   true,
+		ResourceUpToDate: ec2.IsVpcUpToDate(cr.Spec.ForProvider, observed),
 	}, nil
 }
 
 func (e *external) Create(ctx context.Context, mgd resource.Managed) (managed.ExternalCreation, error) {
-	cr, ok := mgd.(*v1alpha3.VPC)
+	cr, ok := mgd.(*v1beta1.VPC)
 	if !ok {
 		return managed.ExternalCreation{}, errors.New(errUnexpectedObject)
 	}
 
-	// if VPCID already exists, skip creating the vpc
-	// this happens when an error has occurred when modifying vpc attributes
-	if meta.GetExternalName(cr) == "" {
-
-		req := e.client.CreateVpcRequest(&awsec2.CreateVpcInput{
-			CidrBlock: aws.String(cr.Spec.CIDRBlock),
-		})
+	cr.Status.SetConditions(runtimev1alpha1.Creating())
 
-		rsp, err := req.Send(ctx)
-		if err != nil {
-			return managed.ExternalCreation{}, errors.Wrap(err, errCreate)
-		}
-
-		meta.SetExternalName(cr, aws.StringValue(rsp.Vpc.VpcId))
-		if err := e.kube.Update(ctx, cr); err != nil {
-			return managed.ExternalCreation{}, errors.Wrap(err, errPersistExternalName)
-		}
+	result, err := e.client.CreateVpcRequest(&awsec2.CreateVpcInput{
+		CidrBlock:       aws.String(cr.Spec.ForProvider.CIDRBlock),
+		InstanceTenancy: awsec2.Tenancy(aws.StringValue(cr.Spec.ForProvider.InstanceTenancy)),
+	}).Send(ctx)
 
-		cr.SetConditions(runtimev1alpha1.Creating())
-		cr.UpdateExternalStatus(*rsp.Vpc)
+	if err != nil {
+		return managed.ExternalCreation{}, errors.Wrap(err, errCreate)
 	}
 
+	meta.SetExternalName(cr, aws.StringValue(result.Vpc.VpcId))
```

Comment:
Right after setting the external name, we should call `e.kube.Update()` so that the annotation is not lost if the next steps fail for some reason.